src/main/resources/script_templates/Neuroanatomy/pyimagej/pyijtest.py

```diff
@@ -1,67 +1,82 @@
 
-script = """
-# @Context context
-# @LegacyService ls
-# @DatasetService ds
-# @DisplayService display
-# @LogService log
-# @SNTService snt
-# @StatusService status
-# @UIService ui
-# @output String dendrite_tip_list
-# @output String axon_tip_list
-
-
-from tracing import (Path, PathAndFillManager, SimpleNeuriteTracer, SNTUI, Tree)
-from tracing.io import MLJSONLoader
-from tracing.util import PointInImage
-from tracing.analysis import (RoiConverter, TreeAnalyzer, TreeColorMapper, 
-    TreeStatistics)
-from tracing.viewer import(Viewer2D, Viewer3D)
+# import relevant Java classes
+HashSet = autoclass('java.util.HashSet')
+PointInImage = autoclass('tracing.util.PointInImage')
+MouseLightLoader = autoclass('tracing.io.MouseLightLoader')
+Tree = autoclass('tracing.Tree')
+TreeAnalyzer = autoclass('tracing.analysis.TreeAnalyzer')
+Color = autoclass('org.scijava.util.Colors')
+Viewer = autoclass('tracing.viewer.Viewer3D')
 
 
 def run():
-
-    # Import some data from the MouseLight database in 'headless' mode
-    loader = MLJSONLoader("AA0100") # one of the largest cells in the database
+    # fetch swc from MouseLight database by ID
+    loader = MouseLightLoader('AA0265')
     if not loader.isDatabaseAvailable():
-        ui.showDialog("Could not connect to ML database", "Error")
+        print("Could not connect to ML database", "Error")
         return
     if not loader.idExists():
-        ui.showDialog("Somewhow the specified id was not found", "Error")
+        print("Somewhow the specified id was not found", "Error")
         return
 
-    d_tree = loader.getTree('dendrites', None)  # compartment, color
-    a_tree = loader.getTree('axon', None)
-       
-    dendrite_tip_positions = TreeAnalyzer(d_tree).getTips()
-    dendrite_tip_list = [[tip.x, tip.y, tip.z] for tip in dendrite_tip_positions]
-    
-    axon_tip_positions = TreeAnalyzer(a_tree).getTips()
-    axon_tip_list = [[tip.x, tip.y, tip.z] for tip in axon_tip_positions]
-    
-    return axon_tip_list, dendrite_tip_list
-    
-axon_tip_list, dendrite_tip_list = run()
-"""
+    tree = loader.getTree('axon', None)
+    analyzer = TreeAnalyzer(tree)
+
+    # extract the axonal end points from the tree as a java HashSet
+    tips_java_set = analyzer.getTips()
+    tips_iterator = tips_java_set.iterator()
+
+    # convert to python list
+    tips_list = []
+    while tips_iterator.hasNext():
+        n = tips_iterator.next()
+        tips_list.append([n.x, n.y, n.z])
+
+    assert len(tips_list) == tips_java_set.size()
+
+    # Find the convex hull of the end points
+    X = np.asarray(tips_list)
+    hull = ConvexHull(X)
+    print("The volume of the convex hull encapsulating all axonal end points is {} cubic microns".format(
+        round(hull.volume, 2)))
+    print("The area of the hull is {} square microns".format(round(hull.area, 2)))
+
+    verts = X[hull.vertices]
+
+    # construct new java Hashset containing the hull vertices
+    verts_set = HashSet()

```

Comment:
You can write `ij.py.to_java(set(verts))` to go from Python set to Java set.

Reply:
Hi @ctrueden thank you for the tips. I suppose the difficulty lies in converting between interfaces specific to Fiji, in this case SNTPoint, and python data types. From my limited perusing, I wasn't able to find a way to do this within scyjava. If this is already supported please let me know.

A couple alternative options come to mind:

option 1.
```
tips_java_set = analyzer.getTips()
tips_java_array = tips_java_set.toArray()
```
then use ``` hull.vertices ``` to index the specific vertices from ``` tips_java_array ```.  I am not sure whether accessing java arrays by index bypass the calls over JNI, but it would forgo having to construct a new PointInImage() Java class for each computed vertex.

option 2:
add a method within SNT to accept a collection of ArrayLists and convert into a SNTPoint set. 

@tferr thoughts?

Reply:
option 3: implement the SNTPoint interface from within python using jnius.PythonJavaClass

Reply:
would option 3 provide a performance gain over using autoclass(tracing.util.PointInImage)?